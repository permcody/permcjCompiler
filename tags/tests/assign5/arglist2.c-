int kate(int is, bool bs, int ia[], bool ba[])
{
	return 666;
}

int matty(void)
{
    int inta[2];
    bool boola[3];

    kate(101);
    kate(101, true);
    kate(101, true, inta);
    kate(101, true, inta, boola);
    kate(101, true, inta, boola, 107);
    kate(101, true, inta, boola, 107, 109);

    kate(true, inta, boola, 101);
    kate(inta, boola, 101, true);
    kate(boola, 101, true, inta);
    kate(101, true, inta, boola);

    kate(101+true, true, inta, boola);

    return 42;
}

int abbey(void)
{
	return;
}

int abbey(void)
{
	return;
}

int zena(void)
{
	return true;
}

bool zeus(void)
{
	return 666;
}

bool blackdog(void)
{
	return;
}

int zog(void)
{
	int z[2];

	return z;
}

void quaoar(void)
{
	return 666;
}

void ixion(void)
{
	return true;
}

void varuna(void)
{
	bool z[10];
	return z;
}
