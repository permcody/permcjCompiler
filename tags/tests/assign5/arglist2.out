ERROR(11): Too few parameters passed for function 'kate' defined on line 1.
ERROR(12): Too few parameters passed for function 'kate' defined on line 1.
ERROR(13): Too few parameters passed for function 'kate' defined on line 1.
ERROR(15): Too many parameters passed for function 'kate' defined on line 1.
ERROR(16): Too many parameters passed for function 'kate' defined on line 1.
ERROR(18): Expecting type int in parameter 1 of call 'kate' defined on line 1.
ERROR(18): Expecting type bool in parameter 2 of call 'kate' defined on line 1.
ERROR(18): Not expecting array in parameter 2 of call 'kate' defined on line 1.
ERROR(18): Expecting type int in parameter 3 of call 'kate' defined on line 1.
ERROR(18): Expecting type bool in parameter 4 of call 'kate' defined on line 1.
ERROR(18): Expecting array in parameter 4 of call 'kate' defined on line 1.
ERROR(19): Not expecting array in parameter 1 of call 'kate' defined on line 1.
ERROR(19): Not expecting array in parameter 2 of call 'kate' defined on line 1.
ERROR(19): Expecting array in parameter 3 of call 'kate' defined on line 1.
ERROR(19): Expecting array in parameter 4 of call 'kate' defined on line 1.
ERROR(20): Expecting type int in parameter 1 of call 'kate' defined on line 1.
ERROR(20): Not expecting array in parameter 1 of call 'kate' defined on line 1.
ERROR(20): Expecting type bool in parameter 2 of call 'kate' defined on line 1.
ERROR(20): Expecting type int in parameter 3 of call 'kate' defined on line 1.
ERROR(20): Expecting array in parameter 3 of call 'kate' defined on line 1.
ERROR(20): Expecting type bool in parameter 4 of call 'kate' defined on line 1.
ERROR(23): '+' requires things of type int but rhs is of type bool.
ERROR(30): Expecting a return type of int but got none.
ERROR(33): Symbol 'abbey' is already defined at line 28.
ERROR(35): Expecting a return type of int but got none.
ERROR(40): Expecting return type of int but got type bool.
ERROR(45): Expecting return type of bool but got type int.
ERROR(50): Expecting a return type of bool but got none.
ERROR(57): Cannot return an array.
ERROR(62): Expecting no return value, but got one.
ERROR(67): Expecting no return value, but got one.
ERROR(73): Expecting no return value, but got one.
ERROR(73): Cannot return an array.
ERROR(LINKER): Procedure main is not defined.
Number of warnings: 0
Number of errors: 34
